src/services/branches/useBranchBundles.tsx

```diff
@@ -0,0 +1,172 @@
+import { useQuery } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import { MissingHeadReportSchema } from 'services/comparison'
+import {
+  RepoNotFoundErrorSchema,
+  RepoOwnerNotActivatedErrorSchema,
+  useRepoOverview,
+} from 'services/repo'
+import Api from 'shared/api'
+import A from 'ui/A'
+
+const BundleSchema = z.object({
+  name: z.string(),
+  sizeTotal: z.number(),
+  loadTimeTotal: z.number(),
+})
+
+const BundleAnalysisReportSchema = z.object({
+  __typename: z.literal('BundleAnalysisReport'),
+  sizeTotal: z.number(),
+  loadTimeTotal: z.number(),
+  bundles: z.array(BundleSchema),
+})
+
+const BundleReportSchema = z.discriminatedUnion('__typename', [
+  BundleAnalysisReportSchema,
+  MissingHeadReportSchema,
+])
+
+const RepositorySchema = z.object({
+  __typename: z.literal('Repository'),
+  branch: z
+    .object({
+      head: z
+        .object({
+          bundleAnalysisReport: BundleReportSchema.nullable(),
+        })
+        .nullable(),
+    })
+    .nullable(),
+})
+
+const BranchBundleSummaryDataSchema = z.object({
+  owner: z
+    .object({
+      repository: z
+        .discriminatedUnion('__typename', [
+          RepositorySchema,
+          RepoNotFoundErrorSchema,
+          RepoOwnerNotActivatedErrorSchema,
+        ])
+        .nullable(),
+    })
+    .nullable(),
+})
+
+const query = `query BranchBundleSummaryData(
+  $owner: String!
+  $repo: String!
+  $branch: String!
+) {
+  owner(username: $owner) {
+    repository(name: $repo) {
+      __typename
+      ... on Repository {
+        branch(name: $branch) {
+          head {
+            bundleAnalysisReport {
+              __typename
+              ... on BundleAnalysisReport {
+                sizeTotal
+                loadTimeTotal
+                bundles {
+                  name
+                  sizeTotal
+                  loadTimeTotal
+                }
+              }
+              ... on MissingHeadReport {
+                message
+              }
+            }
+          }
+        }
+      }
+      ... on NotFoundError {
+        message
+      }
+      ... on OwnerNotActivatedError {
+        message
+      }
+    }
+  }
+}`
+
+export interface UseBranchBundleSummaryArgs {
+  provider: string
+  owner: string
+  repo: string
+  branch?: string
+}
+
+export const useBranchBundleSummary = ({
+  provider,
+  owner,
+  repo,
+  branch: branchArg,
+}: UseBranchBundleSummaryArgs) => {
+  const { data: repoOverview, isLoading } = useRepoOverview({
+    provider,
+    repo,
+    owner,
+  })
+
+  const branch = branchArg ?? repoOverview?.defaultBranch
```

Comment:
what if repo doesn't have a default branch?

Reply:
According to the DB the default branch field on the repos table is not null, and incase the branch doesn't have one it looks like it will set a default value to that field, but just incase I'll add in a check that will continue to disable the query if there isn't a branch.